crates/oxc_minifier/src/peephole/substitute_alternate_syntax.rs

```diff
@@ -939,6 +940,75 @@ impl<'a> LatePeepholeOptimizations {
         Some(ctx.ast.expression_unary(lit.span, UnaryOperator::LogicalNot, num))
     }
 
+    fn try_compress_array_expression(
+        expr: &mut Expression<'a>,
+        ctx: Ctx<'a, '_>,
+    ) -> Option<Expression<'a>> {
+        // this threshold is chosen by hand by checking the minsize output
+        const THRESHOLD: usize = 40;
+
+        let Expression::ArrayExpression(array) = expr else { unreachable!() };
+
+        let is_all_string = array
+            .elements
+            .iter()
+            .all(|element| element.as_expression().is_some_and(Expression::is_string_literal));
+        if !is_all_string {
+            return None;
+        }
+
+        let element_count = array.elements.len();
+        // replace with `.split` only when the saved size is great enough
+        // because using `.split` in some places and not in others may cause gzipped size to be bigger
+        let can_save = element_count * 2 > ".split('.')".len() + THRESHOLD;
+        if !can_save {
+            return None;
+        }
```

Comment:
closure compiler always replaces with `.split` when the saving amount is bigger than 0 (same as setting `THRESHOLD = 0` ). On one hand, that reduce the non-gzipped code size. But on the other hand, it increases the gzipped code size in many cases. To prevent that from happening, I introduced the `THRESHOLD` parameter. I manually selected the value `40` by checking the minsize output. This may be overfit to the benchmark, but I guess it's better than 0.